lib/job/fail.js

```diff
@@ -18,5 +19,15 @@ module.exports = function(reason) {
   this.attrs.failedAt = now;
   this.attrs.lastFinishedAt = now;
   debug('[%s:%s] fail() called [%d] times so far', this.attrs.name, this.attrs._id, this.attrs.failCount);
+  const retryCount = this.attrs.failCount - 1;
+  if (retryCount <= this.attrs.maxRetries) {
```

Comment:
I think this is wrong. it should be < and not <= . in my adaptation if you modify your test to maxRetry 0 it would still pass the test even though the test should fail.